src/main/kotlin/com/exactpro/th2/check1/rule/AbstractCheckTask.kt

```diff
@@ -486,5 +489,27 @@ abstract class AbstractCheckTask(
         return sequence ?: DEFAULT_SEQUENCE
     }
 
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+
+        other as AbstractCheckTask
+
+        if (description != other.description) return false
+        if (timeout != other.timeout) return false
+        if (sessionKey != other.sessionKey) return false
+        if (parentEventID != other.parentEventID) return false
+
+        return true
+    }
+
+    override fun hashCode(): Int {
+        var result = description?.hashCode() ?: 0
+        result = 31 * result + timeout.hashCode()
+        result = 31 * result + sessionKey.hashCode()
+        result = 31 * result + parentEventID.hashCode()
+        return result
+    }
```

Comment:
Then why we should use hashCode here https://github.com/th2-net/th2-check1/blob/master/src/main/kotlin/com/exactpro/th2/check1/rule/AbstractCheckTask.kt#L157 ?

Reply:
We use `hashCode` at line 157 to identify the current rule that has produces that line in the log. If we don't override the `hashCode` method it will return a different value for every new instance (even if they have the same timeout, session key, etc). Each rule is unique and the rule should be equal only to itself